### PR TITLE
Js updates 2

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
Display cross products and vectors as described in #95

### CODE REVIEW

File: MJxPrep.js [hunk 1]
```diff
@@ -317,17 +366,66 @@ if (window.MJxPrep) {
...
+    // function calls
+    var processedArgsString = replaceFunctionCalls(argsString, funcName, action)
+    var args = shallowListSplit(processedArgsString)
+
+    // perform the action
+    var finished = expr.substring(0, funcStart) + action(funcName, args)
+    var unfinished = expr.substring(closeCallParens + 1)
+
+    // Recursively process the unfinished string
+    return replaceFunctionCalls(finished + unfinished, funcName, action, finished.length + 1)
```
Comment (ChristopherChudzicki) [inline MJxPrep.js]: Earlier implementation of `replaceFunctionCalls` could be made to enter an infinite loop (see below). This fixes that.

```js
            //0123456890
const expr = 'cat(1) + 1'
const action = (funcName, args) => 'gonna crash cat(1)'
replaceFunctionCalls(expr, 'cat', action)
```
The original expression closes at position 5. When `replaceFunctionCalls` calls itself the first time, it starts searching at postion 5+1, but that 'cat' has been pushed past that position, so it is encountered again ...

File: MJxPrep.js [hunk 2]
```diff
@@ -50,12 +50,12 @@ if (window.MJxPrep) {
     // Replace fact(...) -> {:factAsciiMath((...)):}, with inner parentheses added as necessary
```
Comment (jolyonb) [inline MJxPrep.js]: Comment needs fixing now.

File: MJxPrep.js [hunk 3]
```diff
@@ -50,12 +50,12 @@ if (window.MJxPrep) {
...
     eqn = replaceFunctionCalls(eqn, 'fact', function(funcName, args) {
       validateArgsLength(funcName, args, 1)
-      return '{:factAsciiMath(' + groupExpr(args[0]) + '):}'
+      return '{:' + groupExpr(args[0]) + '!:}'
     } )
     // Replace factorial(...) -> {:factAsciiMath((...)):}, with inner parentheses added as necessary
```
Comment (jolyonb) [inline MJxPrep.js]: And here too.

**Comment by jolyonb** [comment]
Sorry for being picky!

**Comment by jolyonb** [comment]
Does this complete our todo list on javascript?

**Comment by ChristopherChudzicki** [comment]
> Does this complete our todo list on javascript?

I think it completes #95, but I'm going to go resolve #117

**Pull request merged**